src/onnx/onnx.cpp

```diff
@@ -548,51 +600,34 @@ struct onnx_parser
                         "PARSE_CONV: auto_pad and padding cannot be specified simultaneously");
                 }
             }
+            op.padding.clear();
             copy(info.attributes["pads"].ints(), std::back_inserter(padding));
-            if(padding.size() != 4)
-            {
-                MIGRAPHX_THROW("PARSE_CONV: padding should have 4 values");
-            }
             check_asym_padding(l0, padding, op);
         }
         if(contains(info.attributes, "strides"))
         {
-            copy(info.attributes["strides"].ints(), op.stride.begin());
+            op.stride.clear();
+            copy(info.attributes["strides"].ints(), std::back_inserter(op.stride));
         }
         if(contains(info.attributes, "dilations"))
         {
-            copy(info.attributes["dilations"].ints(), op.dilation.begin());
+            op.dilation.clear();
+            copy(info.attributes["dilations"].ints(), std::back_inserter(op.dilation));
         }
         if(contains(info.attributes, "auto_pad"))
         {
-            auto s = info.attributes["auto_pad"].s();
-            if(s.find("SAME") != std::string::npos)
-            {
-                op.padding_mode                 = op::padding_mode_t::same;
-                std::vector<size_t> weight_dims = weights->get_shape().lens();
-                size_t weight_h                 = weight_dims[2];
-                size_t weight_w                 = weight_dims[3];
-
-                auto input_dims = l0->get_shape().lens();
-                padding.resize(input_dims.size());
-                calculate_padding(
-                    0, padding, input_dims[2], op.stride[0], op.dilation[0], weight_h);
-                calculate_padding(
-                    1, padding, input_dims[3], op.stride[1], op.dilation[1], weight_w);
-
-                check_asym_padding(l0, padding, op);
-            }
+            auto weight_lens = weights->get_shape().lens();
 
-            auto in_lens                      = args[0]->get_shape().lens();
-            auto weight_lens                  = args[1]->get_shape().lens();
-            std::array<std::size_t, 2> k_lens = {weight_lens[2], weight_lens[3]};
+            std::vector<std::size_t> k_lens(weight_lens.begin() + 2, weight_lens.end());
             l0 = process_auto_pad_attribute(l0, info, op, k_lens, op.dilation, in_lens);
         }
         if(contains(info.attributes, "group"))
         {
             op.group = parse_value(info.attributes.at("group")).at<int>();
         }
 
+        recalc_conv_attributes(op, kdims);
```

Comment:
Should we fill the default value or just throw an exception? 
@pfultz2 what is your opinion?

Reply:
We should always fill the default values for padding, strides, and dilations based on K-dims because onnx does not require these attributes when creating the node.

Reply:
For an attribute, I agree that if not value is provided, we should set the default value. But if an incorrect value (less than the required number of dimensions) is provided, we should throw an exception. Looks like the above logic will also pad the missing dimensions with the default value.

So, for each attribute, the logic would be like:
if (contains())
{
    
}
else
{
    std::fill_n();
}

if (op.padding.size() != kdims)
{
    MIGRAPHX_THROW()
}

Reply:
I think instead I will make a function that checks the individual attributes if present in the onnx file. This way it'll be easier to hit code coverage as well.

Reply:
you can check to see the new changes I made to see if this is better. I added an extra test case for coverage as well.